scubagoggles/provider.py

```diff
@@ -685,3 +690,13 @@ def _get_list(resource, item: str, **list_args) -> list:
             request = resource.list_next(request, response)
 
         return results
+
+    def _check_scopes(self, exc):
+        # If one of the scopes is not authorized in a Service account the
+        # error is thrown: ('access_denied: Requested client not authorized.',
+        # {'error': 'access_denied', 'error_description': 'Requested client not authorized.'})
+        scopes_list = self._credentials.scopes
+        if 'access_denied: Requested client not authorized.' in str(exc):
+            logging.error(f'Your credential may be missing one'
+                          f' of the following scopes: {scopes_list}')
+        raise
```

Comment:
Shouldn't this `raise` be inside the `if`? If it's outside, it will throw the exception even if the error is unrelated to the scopes.

For example, the second check scopes call in the screenshot.
![image](https://github.com/user-attachments/assets/d0d881e6-d4a6-44ab-8c85-7a3f9416f071)

If an exception is thrown related to the scopes, it will be caught by the `RefreshError` handler. But if a different kind of exception is thrown, it will be caught by the generic handler. That generic handler calls the check scopes function, then gets interupted early because that function re-raises the exception even though it's not related to the scopes.

```suggestion
            raise
```